prow/oss/terraform/modules/alerts/main.tf

```diff
@@ -12,6 +12,44 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+
+resource "google_monitoring_alert_policy" "ci-absent" {
+  for_each     = var.prow_components
+  project      = var.project
+  display_name = "ci-absent-${each.key}"
+  combiner     = "OR" # required
+
+  conditions {
+    display_name = "ci-absent-${each.key}"
+
+    condition_monitoring_query_language {
+      duration = "1800s"
+      query    = <<-EOT
+      fetch k8s_container
+      | metric 'kubernetes.io/container/restart_count'
+      | filter
+          (resource.container_name == '${each.key}' && resource.namespace_name == '${each.value.namespace}')
+      | align rate(5m)
+      | every 5m
+      | group_by [], [row_count: row_count()]
+      | condition val() > 10 '1'
+      EOT
+      trigger {
+        count = 1
+      }
+    }
+  }
+
+  documentation {
+    content   = "The service `${each.key}` has been restarting more than 10 times within 30 minutes."
```

Comment:
I think this is still inaccurate. Now were checking if there are 10 restarts per every 5mins for a 30min time span. This is a bit confusing, is a long alerting delay, and may still not fire due to crash loop backoff.  How about alerting if a service restarts more than 3 times in 5 mins?

Reply:
3 times in 5 mins will cause false positive, especially when cluster master is upgrading, some components would crashlooping for 5-10 minutes